static/js/components/AnnotationsTable.jsx

```diff
@@ -90,15 +91,27 @@ const AnnotationsTable = ({ annotations, spectrumAnnotations = [] }) => {
   };
 
   const [isRemoving, setIsRemoving] = useState(null);
-  const handleDelete = async (obj_id, id) => {
-    setIsRemoving(id);
-    await dispatch(sourceActions.deleteAnnotation(obj_id, id));
+  const handleDelete = async (id, spectrum_id, annotation_id, type) => {
+    setIsRemoving(annotation_id);
+    if (type === "source") {
+      await dispatch(sourceActions.deleteAnnotation(id, annotation_id));
+    } else if (type === "spectrum") {
+      await dispatch(
+        spectraActions.deleteAnnotation(spectrum_id, annotation_id)
+      );
+    }
     setIsRemoving(null);
   };
 
   // Curate data
+  annotations?.forEach((annotation) => {
+    annotation.type = "source";
+  });
+  spectrumAnnotations?.forEach((spectrumAnnotation) => {
+    spectrumAnnotation.type = "spectrum";
+    annotations?.push(spectrumAnnotation);
+  });
```

Comment:
I'm not sure this is the best place to mark these. Maybe in the duck that puts them into "annotations" and "spectrumAnnotations"? That way you will have that "type" in any component.

Reply:
@guynir42 to confirm: the handlers right?

Reply:
In either the handlers or in the duck where you change the store's state, if you want this to only exist in the front end.

Reply:
I added it to the handlers.